### PR TITLE
Add autoscaling API skelton

### PR DESCRIPTION
**Pull request by jasontedor**
The main purpose of this commit is to add a single autoscaling REST endpoint skeleton, for the purpose of starting to build out the build and testing infrastructure that will surround it. For example, rather than commiting a fully-functioning autoscaling API, we introduce here the skeleton so that we can start wiring up the build and testing infrastructure, establish security roles/permissions, an so on. This way, in a forthcoming PR that introduces actual functionality, that PR will be smaller and have less distractions around that sort of infrastructure.

### CODE REVIEW

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public class GetAutoscalingDecisionAction extends ActionType<GetAutoscalingDecisionAction.Response> {
+
+    public static final GetAutoscalingDecisionAction INSTANCE = new GetAutoscalingDecisionAction();
+    public static final String NAME = "cluster:admin/xpack/autoscaling/get_autoscaling_decision";
```
Comment (henningandersen) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java]: This is probably fine, only wanted to be sure we want `xpack` in the name. Seems most xpack components have that, except ilm, slm and transform, so I am guessing yes?

Comment (henningandersen) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java]: I would opt to remove the second autoscaling in the name, i.e., just:

```suggestion
    public static final String NAME = "cluster:admin/xpack/autoscaling/get_decision";
```

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+        }
+
+        public Response(final StreamInput in) throws IOException {
+            super(in);
+        }
+
+        @Override
+        public void writeTo(final StreamOutput out) {
+
```
Comment (henningandersen) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java]: nit: whitespace

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            builder.endObject();
+            return builder;
+        }
+
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        public Response() {
+
```
Comment (henningandersen) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java]: nit: whitespace

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            builder.startObject();
+            {
+
+            }
+            builder.endObject();
+            return builder;
+        }
+
+    }
+
```
Comment (henningandersen) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java]: nit: whitespace

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        @Override
+        public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
+            builder.startObject();
+            {
+
+            }
+            builder.endObject();
+            return builder;
+        }
+
```
Comment (henningandersen) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java]: nit: whitespace

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/TransportGetAutoscalingDecisionAction.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        final ActionListener<GetAutoscalingDecisionAction.Response> listener
+    ) {
+        listener.onResponse(new GetAutoscalingDecisionAction.Response());
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(final GetAutoscalingDecisionAction.Request request, final ClusterState state) {
+        return null;
+    }
+
```
Comment (henningandersen) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/TransportGetAutoscalingDecisionAction.java]: nit: whitespace

**Review by henningandersen — APPROVED** [review]
LGTM.

I left a few "nit: whitespace", please disregard these. This is only my first real exposure to auto-formatting and I will ask Rory separately about this.

File: x-pack/docs/en/rest-api/security/get-builtin-privileges.asciidoc [hunk 7]
```diff
@@ -66,6 +66,7 @@ A successful call returns an object with "cluster" and "index" fields.
     "delegate_pki",
     "manage",
     "manage_api_key",
+    "manage_autoscaling",
```
Comment (henningandersen) [inline x-pack/docs/en/rest-api/security/get-builtin-privileges.asciidoc]: I think this leaks a bit of docs on autoscaling into release builds. I think this is fair (completely hiding it is probably too much trouble), but maybe we need a short alternative autoscaling docs description explaining this (in a follow-up)?

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java (hunk 1 above)
Comment (jasontedor) [reply x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java]: I'm good leaving it out. I pushed 0db163aee3bf5192bd9e786e3a1116b30bcf2b3b.

**Pull request merged**